antenna/external/fs/crashstorage.py

```diff
@@ -0,0 +1,132 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+import os
+import os.path
+
+from everett.component import ConfigOptions, RequiredConfigMixin
+
+from antenna.util import json_ordered_dumps
+
+
+logger = logging.getLogger(__name__)
+
+
+class FSCrashStorage(RequiredConfigMixin):
+    """Saves raw crash files to the file system
+
+    This generates a tree something like this which mirrors what we do
+    on S3:
+
+    ::
+
+        <FS_ROOT>/
+            <YYYYMMDD>/
+                raw_crash/
+                    <CRASHID>.json
+                dump_names/
+                    <CRASHID>.json
+                <DUMP_NAME>/
+                    <CRASHID>
+
+
+    Couple of things to note:
+
+    1. This doesn't ever delete anything from the tree. You should run another
+       process to clean things up.
+
+    2. If you run out of disk space, this component will fail miserably.
+       There's no way to recover from a full disk--you will lose crashes.
+
+    FIXME(willkg): Can we alleviate or reduce the likelihood of the above?
+
+    """
+    required_config = ConfigOptions()
+    required_config.add_option(
+        'fs_root',
+        default='/tmp/antenna_crashes',
+        doc='path to where files should be stored'
+    )
+
+    # FIXME(willkg): umask
+
+    def __init__(self, config):
+        self.config = config.with_options(self)
+
+        self.root = os.path.abspath(self.config('fs_root'))
+        # FIXME(willkg): We should probably do more to validate fs_root. Remove
+        # / at the end? Verify we can write files to it?
+        if not os.path.isdir(self.root):
+            os.makedirs(self.root)
+
+    def _get_crash_date(self, crash_id):
+        # NOTE(willkg): This assumes we're in the 21st century and that the last
+        # six characters of the crash_id are YYMMDD.
+        return '20' + crash_id[-6:]
```

Comment:
The `antenna.lib.ooid` module has some functions for manipulating crash ids and this should probably be there, but those functions return `datetime`s and that's not what I want. So I could adjust those or add another function, but I figured I'd see if I needed to use that anywhere else first and if so, then centralize it.